### PR TITLE
Add zetainteractor and errors to examples

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary

- Add ZetaInteractor interface to all example contracts
- Update require to solidity errors in example contracts

### CODE REVIEW

**Review by lucas-janon — COMMENTED** [review]
I think I'd go for removing `_crossChainId` from the examples and only using `ZetaInteractor`. To know which is the destination chain we can send `chainId` as an argument in the cross-chain functions and do the `_isValidChainId` check using that argument.

**Comment by andresaiello** [comment]
> _crossChainId

I like it. Already implemented

File: packages/example-contracts/contracts/cross-chain-counter/CrossChainCounter.sol [hunk 1]
```diff
@@ -2,43 +2,29 @@
...
-
-    function setCrossChainId(uint256 ccId) public onlyOwner {
-        _crossChainId = ccId;
-    }
+    constructor(address connectorAddress_) ZetaInteractor(connectorAddress_) {}
 
-    function crossChainCount() external {
-        require(_crossChainAddress.length != 0, "Cross-chain address is not set");
-        require(_crossChainId != 0, "Cross-chain id is not set");
+    function crossChainCount(uint256 crossChainId) external {
```
Comment (lucas-janon) [inline packages/example-contracts/contracts/cross-chain-counter/CrossChainCounter.sol:21]: Can we change this one to `destinationChainId`? I think it'll make the API more clear.

File: packages/example-contracts/contracts/cross-chain-message/CrossChainMessage.sol [hunk 2]
```diff
@@ -2,55 +2,33 @@
...
-     */
-    function setCrossChainId(uint256 _ccId) public onlyOwner {
-        _crossChainId = _ccId;
-    }
+    constructor(address connectorAddress_) ZetaInteractor(connectorAddress_) {}
 
-    function sendHelloWorld() external {
-        require(_crossChainAddress.length != 0, "Cross-chain address is not set");
-        require(_crossChainId != 0, "Cross-chain id is not set");
+    function sendHelloWorld(uint256 crossChainId) external {
```
Comment (lucas-janon) [inline packages/example-contracts/contracts/cross-chain-message/CrossChainMessage.sol:25]: Same for this one `destinationChainId`

**Pull request merged**